### PR TITLE
fix(curriculum): Allow leading to zero before decimal point to be optional for opacity value

### PR DESCRIPTION
**Pull request by RandellDawson**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Per discussion in [this forum topic](https://www.freecodecamp.org/forum/t/applied-visual-design-tutorial-bug-w-opacity/325805), I have modified the test to accept the opacity value with or without the leading zero before the decimal point.  They are both functionality valid syntax.

[Applied Visual Design: Decrease the Opacity of an ElementPassed](https://www.freecodecamp.org/learn/responsive-web-design/applied-visual-design/decrease-the-opacity-of-an-element)

### CODE REVIEW

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

File: curriculum/challenges/english/01-responsive-web-design/applied-visual-design/decrease-the-opacity-of-an-element.english.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ Set the <code>opacity</code> of the anchor tags to 0.7 using <code>links</code>
 ```yml
 tests:
   - text: Your code should set the <code>opacity</code> property to 0.7 on the anchor tags by selecting the class of <code>links</code>.
-    testString: assert(/\.links\s*\{[^}]+opacity\s*:\s*0.7;/.test(code));
+    testString: assert(/\.links\s*\{[^}]+opacity\s*:\s*0?\.7;/.test(code));
```
Comment (moT01) [inline curriculum/challenges/english/01-responsive-web-design/applied-visual-design/decrease-the-opacity-of-an-element.english.md]:
```suggestion
    testString: assert(/\.links\s*\{[^}]*opacity\s*:\s*0?\.7\s*;?\s*}/.test(code));
```

**Review by moT01 — COMMENTED** [review]
I found a few edge cases that we could probably fix quick...

Screen Shot 2019-11-13 at 9 39 55 AM

- It fails if there isn't a space after the opening `{`
- It fails if there is a space before the `;`
- It fails if there isn't a `;` at the end

I think getting the computed style would work here - see suggestion. I tested, and it works with 0.7 and .7

I guess that only checks the first link, and doesn't force the element to be targeted by the class. But maybe something along those lines anyway.

File: curriculum/challenges/english/01-responsive-web-design/applied-visual-design/decrease-the-opacity-of-an-element.english.md (hunk 1 above)
Comment (RandellDawson) [reply curriculum/challenges/english/01-responsive-web-design/applied-visual-design/decrease-the-opacity-of-an-element.english.md]: @moT01 Are we sure all browsers will end up returning the leading zero in the string?

Comment (RandellDawson) [reply curriculum/challenges/english/01-responsive-web-design/applied-visual-design/decrease-the-opacity-of-an-element.english.md]: What if we used the `==` operator instead, just in case?

Comment (moT01) [reply curriculum/challenges/english/01-responsive-web-design/applied-visual-design/decrease-the-opacity-of-an-element.english.md]: I'm not sure all browsers return the same string, it would require some testing. I think I checked on chrome and firefox.

Comment (RandellDawson) [reply curriculum/challenges/english/01-responsive-web-design/applied-visual-design/decrease-the-opacity-of-an-element.english.md]: @moT01 No, I like what you were doing.  I was just wondering if you could change the suggestion to:
```
    testString: assert(getComputedStyle($('.links')[0]).opacity == '0.7');
```
just in case `getComputedStyle` returns `.7` on some browsers.  The `==` operator would handle that case.

Comment (moT01) [reply curriculum/challenges/english/01-responsive-web-design/applied-visual-design/decrease-the-opacity-of-an-element.english.md]: done

**Comment by RandellDawson** [comment]
@ojeytonwilliams @moT01 I made Tom's suggested change.  If you all like, go ahead and merge.

**Review by moT01 — APPROVED** [review]
LGTM @RandellDawson 🎉
